src/model/validation/Mint721BuiltInPrivateKeyValidator.ts

```diff
@@ -0,0 +1,64 @@
+import {
+    isUUID,
+    maxLength,
+    minLength,
+    ValidationArguments,
+    ValidatorConstraint,
+    ValidatorConstraintInterface,
+} from 'class-validator'
+import { Currency } from '../request'
+import { SignatureIdValidator } from './SignatureIdValidator'
+
+@ValidatorConstraint({ name: 'builtInPrivateKey', async: false })
+export class Mint721BuiltInPrivateKeyValidator implements ValidatorConstraintInterface {
+    public defaultMessage(validationArguments?: ValidationArguments) {
+        return 'If you fill signatureId or privateKey/secret/fromPrivateKey, then tokenId, contractAddress must be present. In case of celo feeCurrency must be filled also.'
```

Comment:
feeCurrency use CELO as default if possible

Reply:
What do you mean by that exactly? :-)